File: Distribuidos/Projeto3/cliente.py
import json
import urllib.request
import urllib.error

# Configurações
URL_BASE = "http://localhost:8080"

def enviar_post(endpoint, dados):
    url = f"{URL_BASE}{endpoint}"
    dados_json = json.dumps(dados).encode('utf-8')
    
    req = urllib.request.Request(url, data=dados_json, method='POST')
    req.add_header('Content-Type', 'application/json')
    
    try:
        with urllib.request.urlopen(req) as response:
            resposta = response.read().decode('utf-8')
            return json.loads(resposta)
    except urllib.error.HTTPError as e:
        print(f"Erro {e.code}: {e.read().decode('utf-8')}")
        return None

print("--- Cliente Python: Sistema de Hotel ---")

# 1. Cadastrar Cliente
print("\n1. Cadastrando Cliente via Python...")
payload_cliente = {"nome": "Cliente Python", "cpf": "555.555.555-55"}
cliente = enviar_post("/clientes", payload_cliente)

if cliente:
    print(f"   Sucesso! Cliente cadastrado: ID {cliente['id']}, Nome: {cliente['nome']}")

    # 2. Fazer Reserva
    print("\n2. Tentando reservar quarto 101...")
    payload_reserva = {"cpf": "555.555.555-55", "quarto": 101}
    reserva = enviar_post("/reservas", payload_reserva)
    
    if reserva:
        print(f"   Reserva Realizada! ID: {reserva['idReserva']}, Status: {reserva['status']}")
    else:
        print("   Falha na reserva.")

File: Distribuidos/Projeto3/cliente.js
// cliente.js
const http = require('http');

function enviarPost(path, dados) {
    const data = JSON.stringify(dados);

    const options = {
        hostname: 'localhost',
        port: 8080,
        path: path,
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': data.length
        }
    };

    const req = http.request(options, (res) => {
        let body = '';
        res.on('data', (chunk) => body += chunk);
        res.on('end', () => {
            console.log(`\nRESPOSTA (${path}): Status ${res.statusCode}`);
            console.log("Dados:", body);
        });
    });

    req.on('error', (error) => {
        console.error(error);
    });

    req.write(data);
    req.end();
}

console.log("--- Cliente JavaScript (Node) ---");

// 1. Cadastrar um cliente diferente
setTimeout(() => {
    console.log("Enviando cadastro...");
    enviarPost('/clientes', {
        nome: "Cliente Javascript",
        cpf: "999.999.999-99"
    });
}, 1000);

// 2. Tentar reservar (Quarto diferente, o 201)
setTimeout(() => {
    console.log("Enviando reserva...");
    enviarPost('/reservas', {
        cpf: "999.999.999-99",
        quarto: 201
    });
}, 2000);